ADRs/0011-Python-Path-Resolution.md

```diff
@@ -0,0 +1,306 @@
+# Python Path Resolution
+
+## Status
+PROPOSED
+
+Proposed by: Shams Ul Azeem (15/09/2020)
+
+Discussed with: Adam Gibson
+
+## Context
+Konduit Serving has the ability to run a python script using a PythonStep. There are a bunch of options that need to be setup for the PythonStep which includes, names and data types for inputs & outputs, script locations, python library paths. All of these options remain static from system to system apart from the python library paths variable. Since, every system has a different set of python installations using either conda, virtual environments or regular python installs, it becomes a hassle to manage a PythonStep configuration that's widely usable across different systems. The point of creating this ADR is to specify a simple enough workflow that can take care of the python library paths resolution for us through both using the konduit-serving cli **profiles** and simple configs.  
```

Comment:
Maybe add a bit more about the auto discovery? I'd also add a registration command like konduit register path/to/python/binary kind of like what you do in pycharm.

Reply:
Ah yes! The I'll add the details about the auto discovery. Basically it's going to find the python and conda executables in the PATH environment variable and then assign their respective IDs. 

Also, the registration command is a good idea. It will register and keep it in the python paths even if it's not inside the **PATH** env variable.